src/HonzaBotner.Discord.Services/Commands/PollCommands.cs

```diff
@@ -0,0 +1,95 @@
+    public class PollCommands : BaseCommandModule
+    {
+        private const string PollErrorMessage =
+            "Poll build failed. Make sure the question has less than 256 characters and each option has less than 1024 characters.";
+
+        [GroupCommand]
+        [Description("Creates either yes/no or ABC poll.")]
+        [Priority(1)]
+        public async Task PollCommand(CommandContext ctx,
+            [Description("Options for the pool.")] params string[] options)
+        {
+            if (options.Length == 0)
+            {
+                await ctx.RespondAsync($"You have to ask a question");
+                return;
+            }
+
+            IPoll poll;
+
+            if (options.Length == 1)
+            {
+                poll = new YesNoPoll(ctx.Member.RatherNicknameThanUsername(), ctx.Member.AvatarUrl, options.First());
+            }
+            else if (options.Length - 1 <= AbcPoll.MaxOptions)
+            {
+                poll = new AbcPoll(ctx.Member.RatherNicknameThanUsername(), ctx.Member.AvatarUrl, options.First(),
+                    options.Skip(1).ToList());
+            }
+            else
+            {
+                await ctx.RespondAsync($"Too many options, maximum number of responses is ${AbcPoll.MaxOptions}.");
+                return;
+            }
+
+            try
+            {
+                await poll.PostAsync(ctx.Client, ctx.Channel);
+                await ctx.Message.DeleteAsync();
+            }
+            catch
+            {
+                await ctx.RespondAsync(PollErrorMessage);
+            }
+        }
+
+        [Command("yesno")]
+        [Description("Creates a yesno pool.")]
+        [Priority(2)]
+        public async Task YesnoPollCommand(CommandContext ctx,
+            [RemainingText, Description("Poll's question.")]
+            string question)
+        {
+            try
+            {
+                await new YesNoPoll(ctx.Member.RatherNicknameThanUsername(), ctx.Member.AvatarUrl, question)
+                    .PostAsync(ctx.Client, ctx.Channel);
+                await ctx.Message.DeleteAsync();
+            }
+            catch
+            {
+                await ctx.RespondAsync(PollErrorMessage);
+            }
+        }
+
+        [Command("abc")]
+        [Description("Creates an abc pool.")]
+        [Priority(2)]
+        public async Task AbcPollCommand(CommandContext ctx,
+            [Description("Poll's question.")] string question,
+            [Description("Poll's options.")] params string[] answers)
+        {
+            try
+            {
+                await new AbcPoll(ctx.Member.RatherNicknameThanUsername(), ctx.Member.AvatarUrl, question, answers.ToList())
+                    .PostAsync(ctx.Client, ctx.Channel);
+                await ctx.Message.DeleteAsync();
+            }
+            catch
+            {
+                await ctx.RespondAsync(PollErrorMessage);
+            }
```

Comment:
Log exception

Reply:
Will do once I/someone implements Bot exceptions, so this might wait a bit?

Reply:
Ok that seems reasonabel